lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -1517,8 +1517,9 @@ bool LifetimeChecker::diagnoseMethodCall(const DIMemoryUse &Use,
     // the generic error that we would emit before.
     //
     // That is the only case where we support pattern matching a release.
-    if (Release && AI &&
-        !AI->getSubstCalleeType()->getExtInfo().hasGuaranteedSelfParam())
+    if (Release && AI /*
+        && (!AI->getSubstCalleeType()->hasSelfParam()
+            || !AI->getSubstCalleeType()->getSelfParameter().isGuaranteed())*/)
```

Comment:
@gottesmm Do you know what this `hasGuaranteedSelfParam` check was trying to accomplish? Removing doesn't appear to have any effect on the test suite, and this was the only remaining use of a broken `SILFunctionType::hasGuaranteedSelfParam()` method.

Reply:
I would just rip it out.